cmd/controller/main.go

```diff
@@ -75,6 +76,7 @@ func main() {
 
 	// Set up controller runtime controller
 	cloudProvider := registry.NewCloudProvider(ctx, cloudprovider.Options{ClientSet: clientSet})
+	cloudProvider = cpmetrics.PublishLatency(cloudProvider)
```

Comment:
Semantically, I think this is decorating a cloud provider with cloud provider metrics. 
```suggestion
	cloudProvider = cpmetrics.Decorate(cloudProvider)
```

Reply:
`Decorate` is very generic.  For something more descriptive, how about `DecorateWithLatency`?

Reply:
This is of course, not blocking, but...

When I think about semantically what this, I think of a "Cloud Provider Metrics Decorator". Following the standard go package name resolution, this becomes cloudprovidermetrics.Decorate(cloudprovider), which returns a "Cloud Provider Metrics Decorator". 

I'm not sure there's anything fundamentally related to Latency. e.g. we could start recording counts, and I'd hope we wouldn't need to change the name of this method.